core/trino-main/src/main/java/io/trino/sql/planner/planprinter/anonymize/ExchangeNodeRepresentation.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.sql.planner.planprinter.anonymize;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.trino.sql.planner.OrderingScheme;
+import io.trino.sql.planner.PartitioningHandle;
+import io.trino.sql.planner.Symbol;
+import io.trino.sql.planner.plan.ExchangeNode;
+import io.trino.sql.planner.plan.PlanNodeId;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class ExchangeNodeRepresentation
+        extends AnonymizedNodeRepresentation
+{
+    private final ExchangeNode.Type type;
+    private final ExchangeNode.Scope scope;
+    private final PartitioningHandle partitioningHandle;
```

Comment:
We shouldn't rely on `PartitioningHandle` directly, the fields contained in it can change. I think we just need ConnectorPartitioningHandle from it but we need to anonymise it somehow as it can contain column names depending on what the connector does.

Reply:
> I think we just need ConnectorPartitioningHandle from it but we need to anonymise it somehow as it can contain column names depending on what the connector does.

I think this is not possible at least in `main` because we can't cast it to connector level info.

Reply:
We could extract what we can out of PartitioningHandle->ConnectorPartitioningHandle like the class name and the details if it is instance of SystemPartitioningHandle.

Reply:
ping